src/DiffSharp.Core/DiffSharp.fs

```diff
@@ -287,212 +287,209 @@ type DiffSharp with
+    static member numlaplacian epsilon f x = dsharp.numflaplacian epsilon f x |> snd
     static member numfcurl epsilon f x =
-        let fx, j = DiffSharp.numfjacobian epsilon f x
+        let fx, j = dsharp.numfjacobian epsilon f x
         if j.shape <> [|3; 3|] then failwithf "f must be a function with a three-by-three Jacobian"
-        fx, DiffSharp.stack([j.[2, 1] - j.[1, 2]; j.[0, 2] - j.[2, 0]; j.[1, 0] - j.[0, 1]])
-    static member numcurl epsilon f x = DiffSharp.numfcurl epsilon f x |> snd
+        fx, dsharp.stack([j.[2, 1] - j.[1, 2]; j.[0, 2] - j.[2, 0]; j.[1, 0] - j.[0, 1]])
+    static member numcurl epsilon f x = dsharp.numfcurl epsilon f x |> snd
     static member numfdivergence epsilon f x =
-        let fx, j = DiffSharp.numfjacobian epsilon f x
+        let fx, j = dsharp.numfjacobian epsilon f x
         if j.shape.[0] <> j.shape.[1] then failwithf "f must have a square Jacobian"
         fx, j.trace()
-    static member numdivergence epsilon f x = DiffSharp.numfdivergence epsilon f x |> snd
+    static member numdivergence epsilon f x = dsharp.numfdivergence epsilon f x |> snd
     static member numfcurldivergence epsilon f x =
-        let fx, j = DiffSharp.numfjacobian epsilon f x
+        let fx, j = dsharp.numfjacobian epsilon f x
         if j.shape <> [|3; 3|] then failwithf "f must be a function with a three-by-three Jacobian"
-        fx, DiffSharp.stack([j.[2, 1] - j.[1, 2]; j.[0, 2] - j.[2, 0]; j.[1, 0] - j.[0, 1]]), j.trace()
-    static member numcurldivergence epsilon f x = let _, c, d = DiffSharp.numfcurldivergence epsilon f x in c, d
+        fx, dsharp.stack([j.[2, 1] - j.[1, 2]; j.[0, 2] - j.[2, 0]; j.[1, 0] - j.[0, 1]]), j.trace()
+    static member numcurldivergence epsilon f x = let _, c, d = dsharp.numfcurldivergence epsilon f x in c, d
 
 
 // Functional numerical differentiation API shorthand names
-type DiffSharp with
-    static member numgvp f x v = DiffSharp.numgradv f x v
-    static member numg f x = DiffSharp.numgrad f x
-    static member numhvp f x v = DiffSharp.numhessianv f x v
-    static member numh f x = DiffSharp.numhessian f x
-    static member numgh f x = DiffSharp.numgradhessian f x
-    static member numjvp f x v = DiffSharp.numjacobianv f x v
-    static member numj f x = DiffSharp.numjacobian f x
-    static member numfgvp f x v = DiffSharp.numfgradv f x v
-    static member numfg f x = DiffSharp.numfgrad f x
-    static member numfhvp f x v = DiffSharp.numfhessianv f x v
-    static member numfh f x = DiffSharp.numfhessian f x
-    static member numfgh f x = DiffSharp.numfgradhessian f x
-    static member numfjvp f x v = DiffSharp.numfjacobianv f x v
-    static member numfj f x = DiffSharp.numfjacobian f x    
-
-
-type dsharp = DiffSharp
\ No newline at end of file
+type dsharp with
+    static member numgvp f x v = dsharp.numgradv f x v
+    static member numg f x = dsharp.numgrad f x
```

Comment:
This all feels like we're abbreviation abbreviations, it feels a bit to much to be honest 

Perhaps there should be abbreviations and full naming and no half-abbreviations.

Reply:
I think these names and api design can be revised as we talked about some time ago. For example, instead of the `num` prefix in these numerical differentiation operations, we can have a separate namespace, etc. I think it's worth handling in a dedicated pull request.

Reply:
That's not really an objection, just a comment. You might want to get this part of the API assessed by a target expert currently using torch? It's just very information-dense and I think people (esp those just doing tensor programming, not writing optimizers) might be overwhelmed by the mass of abbreviations and possibilities here

I mean how important are the abbreviations anyway? There are only ever a couple of calls to these functions per optimizer right? And they are crucial calls. It seems an area where people spell might benefit from full names.

Reply:
(it's fine for now, just keep this in mind)

Reply:
I think if things are well-documented in DiffSharp docs and tutorials, I think it's fine to have these abbreviated aliases in addition to the "classical" DiffSharp API names like jacobianv, gradhessian, etc. 

I'm not a fan myself, but things like `jvp`, `hvp` etc. are here to stay. This is mostly gaining popularity due to Google projects.

Reply:
I think we can separate these functional autodiff apis or abbreviated versions to another location other than `dsharp.` so that regular users don't get too exposed to these in intellisense, etc. But personally I think it's also fine if they are exposed. They can discover, read the docs, learn a few things about the capabilities etc.